### PR TITLE
CAD-438 replacement for Prometheus adaptor

### PR DESCRIPTION
**Pull request by CodiePP**
description
-----------

- [x] replacement for Prometheus adaptor

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [ ] documentation added and created (`cd docs; nix-shell --run make`)
- [x] link to an issue
- [ ] link to an epic
- [ ] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

**Comment by CodiePP** [comment]
the output of "example-complex" looks like this:
```
rts_gc_par_max_bytes_copied 0
rts_gc_num_gcs 158
complex_#aggregation_complex_message_Error_timed_max_ns 1579631860203446000
rts_gc_mutator_wall_ms 46553
rts_gc_gc_cpu_ms 81
rts_gc_par_avg_bytes_copied 0
rts_gc_init_cpu_ms 0
rts_gc_bytes_allocated 143660488
rts_gc_num_bytes_usage_samples 57
rts_gc_current_bytes_slop 95712
complex_#aggregation_complex_message_Error_timed_min_ns 4000
rts_gc_max_bytes_slop 105024
rts_gc_par_tot_bytes_copied 0
rts_gc_cumulative_bytes_used 42737720
ekg_server_timestamp_ms 1579631903510
rts_gc_gc_wall_ms 84
rts_gc_mutator_cpu_ms 177
complex_#aggregation_complex_message_Notice_timed_max_ns 1579631860203357000
rts_gc_peak_megabytes_allocated 3
rts_gc_init_wall_ms 2
rts_gc_max_bytes_used 849520
rts_gc_bytes_copied 11523104
rts_gc_wall_ms 46637
rts_gc_current_bytes_used 842272
complex_#aggregation_complex_message_Notice_timed_min_ns 5000
rts_gc_cpu_ms 258
```

File: plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs [hunk 1]
```diff
@@ -7,39 +7,66 @@
...
+           ]
+
+webhandler :: EKG.Server -> Snap ()
+webhandler ekg = do
+    samples <- liftIO $ sampleAll $ EKG.serverMetricStore ekg
+    writeLBS . toLazyByteString . renderSamples $ HM.toList samples
+    pure ()
 
+renderSamples :: [(Text, Value)] -> Builder
+renderSamples [] = mempty
```
Comment (ksaric) [inline plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs]: This is kind of manual, but I guess for this level of output, it's ok?
It won't be if we start adding more output.

File: plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs [hunk 2]
```diff
@@ -7,39 +7,66 @@
...
+    config = setPort port . setHostname host . setAccessLog ConfigNoLog . setErrorLog ConfigNoLog $ defaultConfig
+    site :: Snap ()
+    site = route [
+             ("/metrics/", webhandler ekg)
+           ]
+
+webhandler :: EKG.Server -> Snap ()
+webhandler ekg = do
+    samples <- liftIO $ sampleAll $ EKG.serverMetricStore ekg
+    writeLBS . toLazyByteString . renderSamples $ HM.toList samples
```
Comment (ksaric) [inline plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs]: You could have indented `renderSamples` inside `webhandler` so people reading know that it's used only there?

File: plugins/backend-ekg/lobemo-backend-ekg.cabal [hunk 3]
```diff
@@ -19,19 +19,19 @@ library
...
                        iohk-monitoring,
                        aeson,
                        async,
+                       bytestring,
                        ekg,
                        ekg-core,
-                       ekg-prometheus-adapter,
-                       prometheus == 2.1.2,
                        safe-exceptions,
+                       snap-core,
```
Comment (ksaric) [inline plugins/backend-ekg/lobemo-backend-ekg.cabal]: I see you are using the alternative here :smirk:

File: plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs (hunk 2 above)
Comment (CodiePP) [reply plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs]: I see.. maybe next time. :-)

File: plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs (hunk 1 above)
Comment (CodiePP) [reply plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs]: thanks for your inputs. I'll make it nicer..

**Pull request merged**